src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,227 @@
+  /** Optional: Used to populate the Text component rendered below the input field. */
+  formHelperText?: string | JSX.Element;
+  /** A data object containing key/value pairs that will be added to the form
+   * field submitted data. */
+  hiddenFields?: any;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /* Optional value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** A callback handler function that will be called when the form is submitted.
+   * The function provided must expect an object of key:value pairs. This object will
+   * include {email}, being the user's input, plus any objects passed via the hiddenValues prop.
+   */
+  onSubmit: (values: { [key: string]: string }) => any;
+  /** Used to populate the `<h3>` header title. */
+  title: string;
+  /** Used to specify what is displayed in the component content area. */
+  view?: NewsletterSignupViewType;
+}
+
+/**
+ * The NewsletterSignup component provides a way for patrons to register for an
+ * email-based newsletter distribution list.
+ *
+ * The component can show four different views, depending on the state of the
+ * email submission: form, loading, confirmation, and error.
+ */
+export const NewsletterSignup = chakra(
+  forwardRef<any, NewsletterSignupProps>(
+    (
+      {
+        className,
+        confirmationText = "Thank you! You have successfully subscribed to our email updates! You can update your email subscription preferences at any time using the links at the bottom of the email.",
+        descriptionText = "Stay connected with the latest research news from NYPL, including information about our events, programs, exhibitions, and collections.",
+        formHelperText,
+        hiddenFields,
+        id,
+        isInvalidEmail = false,
+        newsletterSignupType = "whatsOn",
+        onSubmit,
+        title = "Sign Up for Our Newsletter!",
+        view = "form",
+        ...rest
+      },
+      ref?
+    ) => {
+      const viewType = view;
+      const [buttonClicked, setButtonClicked] = useState(false);
+      const [email, setEmail] = useState("");
+
+      // Hook into NYPL breakpoint
+      const { isLargerThanMobile } = useNYPLBreakpoints();
+      const focusRef = useRef<HTMLDivElement>(); // @todo Is this needed? It is a holdover from FeedbackBox
+      const styles = useStyleConfig("NewsletterSignup", {});
+      // The viewType may change as a result of something happening in the internalOnSubmit function.
+      const formView = viewType === "form";
+      const confirmationView = viewType === "confirmation";
+      const errorView = viewType === "error";
+      const iconColor = useColorModeValue(null, "dark.ui.typography.body");
+
+      const internalOnSubmit = (e) => {
+        e.preventDefault();
+        setButtonClicked(true);
```

Comment:
If we have a "loading" view, then the `buttonClicked` state is no longer needed. It is also currently not being updated to `false` and only to `true` so you'd only ever be able to submit the form once.

Reply:
This is fixed